### ISSUE TITLE
Allow `Style/TernaryParentheses` to override `Style/RedundantParentheses` when parentheses are required

### ISSUE THREAD

**Issue by jmuhlfel** [rubocop/rubocop#10648]
## Is your feature request related to a problem? Please describe.

`Style/TernaryParentheses` has infinite loop protection to prevent it from endlessly undoing corrections for `Style/RedundantParentheses` when both cops are enabled. This makes sense, but it has an unfortunate side effect: if `Style/TernaryParentheses` is set to require parentheses (via `EnforcedStyle: require_parentheses` or `require_parentheses_when_complex`), that preference is overridden and no offenses are flagged on ternary expressions that lack parentheses. To illustrate:

```ruby
# rubocop.yml
Style/RedundantParentheses:
  Enabled: true

Style/TernaryParentheses:
  Enabled: true
  EnforcedStyle: require_parentheses_when_complex

# foo.rb
foo = bar.empty? || baz.empty? ? "a" : "b" # no Style/TernaryParentheses offense is registered
```

I figured this out by reading the source for both cops - I don't think it's documented anywhere.

## Describe the solution you'd like

I propose moving the infinite loop protection to `Style/RedundantParentheses`: have it skip flagging offenses on ternary expressions when `Style/TernaryParentheses` is set to require parentheses. That way, rubocop can properly enforce the configurations for both cops.

## Describe alternatives you've considered

The only alternative I'm aware of is to disable `Style/RedundantParentheses`, but it's a good cop to have and I'd prefer to have it enabled.

## Additional context

I've been a rubocop user for years and very much appreciate the work the maintainers do. I'd be glad to put a PR together if you folks are onboard with this idea but don't have the bandwidth.

**Comment by dvandersluis** [comment]
Hi @jmuhlfel thanks for the report, I've opened #10874.

**Comment by jmuhlfel** [comment]
Thanks very much @dvandersluis!
